lib/geom/rectangle.dart

```diff
@@ -0,0 +1,211 @@
+  }
+
+  /// Creates a [Rectangle] providing its position [p] (top left corner) and its dimension [size].
+  ///
+  /// They are provided as [IntPosition]'s intances and converted to double.
+  Rectangle.fromIntPositions(IntPosition p, IntPosition size)
+      : this.fromPositions(p.toPosition(), size.toPosition());
+
+  /// Creates an empty [Rectangle] (that is, width and height equal to zero).
+  ///
+  /// By convention, operations that create empty [Rectangle]s use this to create all of them in the position (0, 0) (as it's irrelevant for empty rectangles).
+  static Rectangle empty() => Rectangle.fromLTWH(0, 0, 0, 0);
+
+  /// Returns whether this is empty or not (either width or height is zero).
+  ///
+  /// By convention, if it's empty, all parameters should be zero (though not mandatory).
+  bool get isEmpty => w == 0 || h == 0;
+
+  /// Returns the x coordinate of the left side of the rectangle.
+  double get left => x;
+
+  /// Returns the x coordinate of the right side of the rectangle.
+  double get right => x + w;
+
+  /// Returns the y coordinate of the top side of the rectangle.
+  double get top => y;
+
+  /// Returns the y coordinate of the bottom side of the rectangle.
+  double get bottom => y + h;
+
+  /// Synonym of [w], returns the width of this rectangle.
+  double get width => w;
+
+  /// Synonym of [h], returns the height of this rectangle.
+  double get height => h;
+
+  /// Returns whether this [Rectangle] overlaps another [Rectangle].
+  ///
+  /// This means that the intersection is non-empty.
+  bool overlapsRectangle(Rectangle other) {
+    return !intersection(other).isEmpty;
+  }
+
+  /// Returns the smallest [Rectangle] fully containing the union of this with the [other] [Rectangle].
+  ///
+  /// This is not a set union! The result of this method does contain the path union, but is an axis-aligned Rectangle, so it might be bigger than the strict union.
+  Rectangle union(Rectangle other) {
+    if (other.isEmpty) {
+      return this;
+    }
+    if (isEmpty) {
+      return other;
+    }
+
+    final double rxi = math.min(left, other.left);
+    final double ryi = math.min(top, other.top);
+
+    final double rxf = math.max(right, other.right);
+    final double ryf = math.max(bottom, other.bottom);
+
+    return Rectangle.fromLTWH(rxi, ryi, rxf - rxi, ryf - ryi);
+  }
+
+  Rectangle intersection(Rectangle other) {
+    if (isEmpty || other.isEmpty) {
+      return empty();
+    }
+
+    final double rxi = math.max(left, other.left);
+    final double ryi = math.max(top, other.top);
+
+    final double rxf = math.min(right, other.right);
+    final double ryf = math.min(bottom, other.bottom);
+
+    if (rxi >= rxf || ryi >= ryf) {
+      return empty();
+    }
+
+    return Rectangle.fromLTWH(rxi, ryi, rxf - rxi, ryf - ryi);
+  }
+
+  Rect asRect() {
+    if (isEmpty) {
+      return const Rect.fromLTWH(0.0, 0.0, 0.0, 0.0);
+    }
+    return Rect.fromLTWH(left, top, width, height);
+  }
+
+  Position toLT() {
```

Comment:
What does `toLT` means?

Reply:
LT=leftTop. that is the position at the left top of the rectangle. it's flutter Rect notation, will add a doc/comment thing